src/openaq/incremental_downloader_parallel.py

```
@@ -62,7 +62,14 @@ def fetch_sensor_pages_parallel_sync(self, sensor_ids: List[int], max_pages_per_
         max_pages_per_sensor = min(max_pages_per_sensor, 16)
         
         for sensor_id in sensor_ids:
-            pages_to_fetch = min(6, max(3, self.client.api.num_keys // len(sensor_ids))) if hasattr(self.client.api, 'num_keys') else 3
+            # Be very conservative with initial parallel fetching
+            # Only fetch 1-2 pages initially to avoid timeouts
+            if len(sensor_ids) >= 6:
+                pages_to_fetch = 1  # Only 1 page for 6+ sensors
+            elif len(sensor_ids) >= 4:
+                pages_to_fetch = 2  # 2 pages for 4-5 sensors
+            else:
+                pages_to_fetch = min(3, max(2, self.client.api.num_keys // (len(sensor_ids) * 4))) if hasattr(self.client.api, 'num_keys') else 2
```

Pylint (3.3.7)

[convention] Line too long (145/100)

(C0301)